pkg/pgmodel/sql_test.go

```diff
@@ -446,136 +519,191 @@ func TestPGXInserterInsertSeries(t *testing.T) {
+		sqlQueries    []sqlQuery
+		metricsGetErr error
 	}{
 		{
 			name: "Zero data",
+			sqlQueries: []sqlQuery{
+				{sql: "CALL _prom_catalog.finalize_metric_creation()"},
+			},
 		},
 		{
 			name: "One data",
-			rows: createRows(1),
+			rows: map[string][]samplesInfo{
+				"metric_0": {{samples: make([]prompb.Sample, 1)}},
+			},
+			sqlQueries: []sqlQuery{
+				{sql: "CALL _prom_catalog.finalize_metric_creation()"},
+				{
+					sql:     "SELECT table_name, possibly_new FROM _prom_catalog.get_or_create_metric_table_name($1)",
+					args:    []interface{}{"metric_0"},
+					results: rowResults{{"metric_0", true}},
+					err:     error(nil),
+				},
+				{
+					sql: `INSERT INTO "prom_data"."metric_0"(time, value, series_id) SELECT * FROM unnest($1::TIMESTAMPTZ[], $2::DOUBLE PRECISION[], $3::BIGINT[]) a(t,v,s) ORDER BY s,t ON CONFLICT DO NOTHING`,
+					args: []interface{}{
+						[]time.Time{time.Unix(0, 0)},
+						[]float64{0},
+						[]int64{0},
+					},
+					results: rowResults{{"metric_0", true}},
+					err:     error(nil),
+				},
+			},
 		},
 		{
 			name: "Two data",
-			rows: createRows(2),
-		},
-		{
-			name:     "Create table error",
-			rows:     createRows(5),
-			queryErr: map[int]error{0: fmt.Errorf("create table error")},
+			rows: map[string][]samplesInfo{
+				"metric_0": {
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+				},
+			},
+			sqlQueries: []sqlQuery{
+				{sql: "CALL _prom_catalog.finalize_metric_creation()"},
+				{
+					sql:     "SELECT table_name, possibly_new FROM _prom_catalog.get_or_create_metric_table_name($1)",
+					args:    []interface{}{"metric_0"},
+					results: rowResults{{"metric_0", true}},
+					err:     error(nil),
+				},
+				{
+					sql: `INSERT INTO "prom_data"."metric_0"(time, value, series_id) SELECT * FROM unnest($1::TIMESTAMPTZ[], $2::DOUBLE PRECISION[], $3::BIGINT[]) a(t,v,s) ORDER BY s,t ON CONFLICT DO NOTHING`,
+					args: []interface{}{
+						[]time.Time{time.Unix(0, 0), time.Unix(0, 0)},
+						[]float64{0, 0},
+						[]int64{0, 0},
+					},
+					results: rowResults{},
+					err:     error(nil),
+				},
+			},
 		},
 		{
-			name:        "Copy from error",
-			rows:        createRows(5),
-			copyFromErr: fmt.Errorf("some error"),
+			name: "Create table error",
+			rows: map[string][]samplesInfo{
+				"metric_0": {
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+				},
+			},
+			sqlQueries: []sqlQuery{
+				{sql: "CALL _prom_catalog.finalize_metric_creation()"},
+				{
+					sql:     "SELECT table_name, possibly_new FROM _prom_catalog.get_or_create_metric_table_name($1)",
+					args:    []interface{}{"metric_0"},
+					results: rowResults{},
+					err:     fmt.Errorf("create table error"),
+				},
+			},
 		},
 		{
-			name:           "Not all data inserted",
```

Comment:
I believe this test case is subsumed under the `Copy from error` one now that we've switched to `INSERT ON UPDATE DO NOTHING`: the only source of errors is an `INSERT` error; if we modify the wrong number of rows that just implies duplicates.